cmd/kubeadm/app/cmd/join.go

```diff
@@ -250,20 +267,25 @@ func (j *Join) Run(out io.Writer) error {
 		}
 	}
 
-	kubeconfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletBootstrapKubeConfigFileName)
+	fmt.Fprintf(out, joinDoneMsgf)
+	return nil
+}
 
-	// Write the bootstrap kubelet config file or the TLS-Boostrapped kubelet config file down to disk
-	if err := kubeconfigutil.WriteToDisk(kubeconfigFile, cfg); err != nil {
-		return err
-	}
+func getTLSBootstrappedClient() (clientset.Interface, error) {
+	fmt.Println("[tlsbootstrap] Waiting for the kubelet to perform the TLS Bootstrap...")
 
-	// Write the ca certificate to disk so kubelet can use it for authentication
-	cluster := cfg.Contexts[cfg.CurrentContext].Cluster
-	err = certutil.WriteCert(j.cfg.CACertPath, cfg.Clusters[cluster].CertificateAuthorityData)
+	kubeletKubeConfig := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletKubeConfigFileName)
+
+	// Loop on every falsy return. Return with an error if raised. Exit successfully if true is returned.
+	err := wait.PollImmediateInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
+		if _, err := os.Stat(kubeletKubeConfig); err == nil {
+			return true, nil
```

Comment:
nit: if we really want to shorten the logic here, we could skip the if clause and write `return (err == nil), nil` here instead.
What do you think is better / more readable?

Reply:
I like the `return (err == nil), nil` one :) done.